streams/buffer.ts

```diff
@@ -232,3 +232,43 @@ export class LimitedTransformStream<T> extends TransformStream<T, T> {
     });
   }
 }
+
+/**
+ * A transform stream that only transforms from the zero-indexed `start` and `end` bytes (both inclusive).
+ *
+ * @example
+ * ```ts
+ * import { RangedByteTransformStream } from "https://deno.land/std@$STD_VERSION/streams/buffer.ts";
+ * const response = await fetch("https://example.com");
+ * const rangedStream = response.body!
+ *   .pipeThrough(new RangedByteTransformStream(3, 8));
+ * ```
+ */
+export class RangedByteTransformStream
+  extends TransformStream<Uint8Array, Uint8Array> {
+  #offset = 0;
+
+  constructor(start: number, end = Infinity) {
+    super({
+      start: () => {
+        end += 1;
+      },
+      transform: (chunk, controller) => {
+        this.#offset += chunk.byteLength;
+        if (this.#offset >= start + 1) {
+          if (start - (this.#offset - chunk.byteLength) >= 0) {
+            chunk = chunk.slice(start - (this.#offset - chunk.byteLength));
+          }
+
+          if (end <= this.#offset) {
+            chunk = chunk.slice(0, chunk.byteLength - (this.#offset - end));
+            controller.enqueue(chunk);
+            controller.terminate();
+          }
+
+          controller.enqueue(chunk);
```

Comment:
Maybe I'm missing something, but this whole block seems slightly off. Should the two separate `if` statements be an `if...else`? Why is `chunk` potentially assigned twice? Why is `controller.enqueue()` possibly called twice depending on the path through the code?